oggm/core/climate.py

```diff
@@ -110,87 +110,39 @@ def process_custom_climate_data(gdir):
-            raise ValueError("Need to set cfg.PATHS['gcm_temp_file']")
-        fpath_temp = cfg.PATHS['gcm_temp_file']
-    if fpath_precc is None:
-        if not ('gcm_precc_file' in cfg.PATHS):
-            raise ValueError("Need to set cfg.PATHS['gcm_precc_file']")
-        fpath_precc = cfg.PATHS['gcm_precc_file']
-    if fpath_precl is None:
-        if not ('gcm_precl_file' in cfg.PATHS):
-            raise ValueError("Need to set cfg.PATHS['gcm_precl_file']")
-        fpath_precl = cfg.PATHS['gcm_precl_file']
-
-    # read the files
-    with warnings.catch_warnings():
-        # Long time series are currently a pain pandas
-        warnings.filterwarnings("ignore", message='Unable to decode time axis')
-        tempds = xr.open_dataset(fpath_temp)
-    precpcds = xr.open_dataset(fpath_precc, decode_times=False)
-    preclpds = xr.open_dataset(fpath_precl, decode_times=False)
-
-    # select for location
-    lon = gdir.cenlon
-    lat = gdir.cenlat
-
-    # CESM files are in 0-360
-    if lon <= 0:
-        lon += 360
-
-    # take the closest
-    # TODO: consider GCM interpolation?
-    temp = tempds.TREFHT.sel(lat=lat, lon=lon, method='nearest')
-    precp = (precpcds.PRECC.sel(lat=lat, lon=lon, method='nearest') +
-             preclpds.PRECL.sel(lat=lat, lon=lon, method='nearest'))
-
-    # from normal years to hydrological years
-    sm = cfg.PARAMS['hydro_month_' + gdir.hemisphere]
-    em = sm - 1 if (sm > 1) else 12
-    # TODO: we don't check if the files actually start in January but we should
-    precp = precp[sm-1:sm-13].load()
-    temp = temp[sm-1:sm-13].load()
-    y0 = int(temp.time.values[0].strftime('%Y'))
-    y1 = int(temp.time.values[-1].strftime('%Y'))
-    time = pd.period_range('{}-{:02d}'.format(y0, sm),
-                           '{}-{:02d}'.format(y1, em), freq='M')
-    temp['time'] = time
-    precp['time'] = time
-    # Workaround for https://github.com/pydata/xarray/issues/1565
-    temp['month'] = ('time', time.month)
-    precp['month'] = ('time', time.month)
-    temp['year'] = ('time', time.year)
-    precp['year'] = ('time', time.year)
-    ny, r = divmod(len(time), 12)
-    assert r == 0
-
-    # Convert m s-1 to mm mth-1
-    ndays = np.tile(np.roll(cfg.DAYS_IN_MONTH, 13-sm), y1 - y0)
-    precp = precp * ndays * (60 * 60 * 24 * 1000)
+    prcp    : xarray.DataArray - format:
+        monthly total precipitation [mm]
+            Coordinates:
+            lat      float64
+            lon      float64
+            time     (time) object
+            month    (time) int64
+            year     (time) int64
+    temp : xarray.DataArray
+        monthly temperature [K]
+            Coordinates:
+            lat      float64
+            lon      float64
+            time     (time) object
+            month    (time) int64
+            year     (time) int64
+    time_unit : str
+            format: days since e.g.: days since 0850-01-01 00:00:00
+    time2 : numpy.ndarray
```

Comment:
what's `time2` here? I think another name and doc would help. Maybe we can even don't use it and extract it from the dataarrays?

Reply:
I agree that it is very unclear in this way, I include the doc and another name. I was thinking maybe `time_dates` would be better. 

`time2` is the variable `time` that is needed in the `write_monthly_climate_file()`. I agree that it would be better if the time could be extracted from the dataarrays. I only don't know yet how to include it.

Reply:
OK. You can give it a try and if not I'll have a look